contrib/corrosion-cmake/CMakeLists.txt

```
@@ -116,3 +120,30 @@ set(RUST_VENDOR_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../rust_vendor")
 # This needs to be done for each project using set_property(INTERFACE_CORROSION_CARGO_FLAGS)
 configure_file("config.toml.in" "config.toml" @ONLY)
 set (RUST_CONFIG_FILE "${CMAKE_CURRENT_BINARY_DIR}/config.toml" CACHE INTERNAL "Rust config")
+
+function(clickhouse_import_crate)
+    # This is a workaround for Corrosion case sensitive build type matching in
+    # _generator_add_cargo_targets(), that leads to different paths in
+    # IMPORTED_LOCATION and real path of the library.
+    #
+    # It uses CMAKE_CONFIGURATION_TYPES and $<CONFIG>, so here we preserve the
+    # case of ${CMAKE_BUILD_TYPE} in ${CMAKE_CONFIGURATION_TYPES}.
+    if ("${CMAKE_BUILD_TYPE_UC}" STREQUAL "DEBUG")
+        set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE};release")
+    else()
+        set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE};debug")
+    endif()
```

Should it be like this?

```suggestion
    if ("${CMAKE_BUILD_TYPE_UC}" STREQUAL "DEBUG")
        set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE};debug")
    else()
        set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE};release")
    endif()
```

---

Good question. I don't know. I copied the duplicated code that already existed but I don't know if it's even necessary anymore.

---

It seems this is related to multi-config cmake. Which we don't use since we are overwritting the generator to Ninja and never to `Ninja Multi-Config`.

Also, AFAIK we never considered multiconfig to be an option and other things rely on one build == one build folder too. Maybe it makes sense to just remove all the `CMAKE_CONFIGURATION_TYPES` config from our CMake files.